docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceApi.php

```diff
@@ -0,0 +1,254 @@
+   *   The Agency Component node object.
+   * @param \Drupal\webform\WebformInterface $webform
+   *   The Webform node object.
+   *
+   * @return array
+   *   Return the assemble request data in an array.
+   */
+  public function assembleRequestData(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent, WebformInterface $webform) {
+    // Get the webform submission values.
+    $formValues = $this->getSubmissionValues($webformSubmission, $webform);
+
+    // Get the agency information.
+    $agencyInfo = $this->getAgencyInfo($agencyComponent);
+
+    $submissionValues = array_merge($formValues, $agencyInfo);
+
+    return $submissionValues;
+  }
+
+  /**
+   * Return the FOIA form submission values as an array.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The Webform submission values.
+   * @param \Drupal\webform\WebformInterface $webform
+   *   The Webform node object.
+   *
+   * @return array
+   *   Returns the submission values as an array.
+   */
+  public function getSubmissionValues(WebformSubmissionInterface $webformSubmission, WebformInterface $webform) {
+    $submissionValues = $webformSubmission->getData();
+    // If there are files attached, load the files and add the file metadata.
+    if ($webform->hasManagedFile()) {
+      $attachments = [];
+      $fileKeys = $this->getFileAttachmentElementsOnWebform($webform);
+      if ($fileKeys) {
+        foreach ($fileKeys as $key) {
+          $files = $this->getAttachmentData($submissionValues[$key]);
+          $attachments = array_merge($files, $attachments);
+          unset($submissionValues[$key]);
+        }
+      }
+      if (!empty($attachments)) {
+        $submissionValues['attachments'] = $attachments;
+      }
+    }
+
+    return $submissionValues;
+  }
+
+  /**
+   * Fetch the Agency information from the Agency Component.
+   *
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   */
+  public function getAgencyInfo(NodeInterface $agencyComponent) {
+    $agencyTerm = $this->agencyLookUpService->getAgencyFromComponent($agencyComponent);
+
+    return [
+      'agency' => ($agencyTerm) ? $agencyTerm->label() : '',
+      'agency_component_name' => $agencyComponent->label(),
+    ];
+  }
+
+  /**
+   * Get the metadata for the file attachments.
+   *
+   * @param array $files
+   *   An array containing the file IDs for the file attachments.
+   */
+  public function getAttachmentData(array $files) {
+    $fileContents = [];
+    if (!empty($files)) {
+      foreach ($files as $fid) {
+        $currentFile = File::load($fid);
+        $base64 = base64_encode(file_get_contents($currentFile->getFileUri()));
+        $fileContents[] = [
+          'content_type' => $currentFile->getMimeType(),
+          'filedata' => $base64,
+          'filename' => $currentFile->getFilename(),
+          'filesize' => $currentFile->getSize(),
```

Comment:
the `getSize` method returns a string, mind if we cast it to an integer?
`(int)$currentFile->getSize();` should do it, but definitely test it out first.